packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1099,6 +1099,13 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  bool _isSelectionOverlayEnabled = true;
```

Comment:
This is just e a editabletext. It shouldn't know so much about whether to show the toolbar based on the platform and pointer type etc etc. Just let its parent call or not call showToolbar() based on the appropriate platform+device information.

Reply:
Agreed. It seems that the logic is still spread across editable text and text field now. I did a clean up and put all the show/hide logic in the text field. PTAL.